app/javascript/components/Popover.tsx

```diff
@@ -1,107 +1,103 @@
+import * as PopoverPrimitive from "@radix-ui/react-popover";
 import * as React from "react";
 
 import { classNames } from "$app/utils/classNames";
 
-import { Details } from "$app/components/Details";
-import { useGlobalEventListener } from "$app/components/useGlobalEventListener";
-import { useOnOutsideClick } from "$app/components/useOnOutsideClick";
+import { ButtonProps, buttonVariants } from "$app/components/Button";
+import { Position, WithTooltip } from "$app/components/WithTooltip";
 
-export type Props = {
-  trigger: React.ReactNode | ((open: boolean) => React.ReactNode);
-  children: React.ReactNode | ((close: () => void) => React.ReactNode);
-  className?: string;
-  dropdownClassName?: string;
-  open?: boolean;
-  onToggle?: (open: boolean) => void;
-  style?: React.CSSProperties;
-  position?: "top" | "bottom" | undefined;
-  "aria-label"?: string;
-  disabled?: boolean;
-};
+export const Popover = PopoverPrimitive.Root;
 
-export const Popover = ({
-  trigger,
-  children,
-  className: triggerClassName,
-  dropdownClassName,
-  open: openProp,
-  onToggle,
-  style,
-  position,
-  "aria-label": ariaLabel,
-  disabled,
-}: Props) => {
-  const [open, setOpen] = React.useState(openProp ?? false);
-  const ref = React.useRef<HTMLElement | null>(null);
+export const PopoverClose = PopoverPrimitive.Close;
 
-  if (openProp !== undefined && open !== openProp) setOpen(openProp);
-
-  const toggle = (newOpen: boolean) => {
-    if (openProp === undefined) setOpen(newOpen);
-    if (newOpen !== open) onToggle?.(newOpen);
-  };
-
-  useOnOutsideClick([ref.current], () => toggle(false));
-  useGlobalEventListener("keydown", (evt) => {
-    if (evt.key === "Escape") {
-      toggle(false);
-    }
-  });
-  const dropoverPosition = useDropdownPosition(ref);
-  React.useEffect(() => {
-    if (!open) return;
-    const focusElement = ref.current?.querySelector("[autofocus]");
-    if (focusElement instanceof HTMLElement) focusElement.focus();
-  }, [open]);
+export const PopoverTrigger = React.forwardRef<
+  React.ElementRef<typeof PopoverPrimitive.Trigger>,
+  React.ComponentPropsWithoutRef<typeof PopoverPrimitive.Trigger> & {
+    tooltip?: string | undefined;
+    tooltipPosition?: Position;
+  }
+>(({ className, tooltip, tooltipPosition, ...props }, ref) => {
+  const content = (
+    <PopoverPrimitive.Trigger
+      ref={ref}
+      className={classNames("outline-none focus-visible:outline-none", className)}
+      {...props}
+    />
+  );
 
-  const renderedTrigger = typeof trigger === "function" ? trigger(open) : trigger;
+  if (tooltip) {
+    return (
+      <WithTooltip position={tooltipPosition} tip={tooltip}>
```

Comment:
I don't think we should move this here in this PR (unless I'm missing some significant difference to styling that requires us to have it here?)

Note: In general I think it's quite a good pattern to have tooltip props to show which elements can reasonably have tooltips, but if we go that route we should do it separately as a tooltip refactor